panic from logmux

```text
now=2015-04-01T19:22:14+0000 app=host backend=libvirt-lxc fn=Cleanup at=finish
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x28 pc=0x59763c]
 
goroutine 7 [running]:
github.com/flynn/flynn/host/logmux.(*LogMux).Close(0xc20804f220)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/host/logmux/logmux.go:95 +0x13c
main.func·005()
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/host/host.go:190 +0x2a
github.com/flynn/flynn/pkg/shutdown.(*handler).exit(0xc2080c01e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/pkg/shutdown/handler.go:62 +0xf7
github.com/flynn/flynn/pkg/shutdown.(*handler).wait(0xc2080c01e0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/pkg/shutdown/handler.go:55 +0x19a
created by github.com/flynn/flynn/pkg/shutdown.newHandler
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/pkg/shutdown/handler.go:25 +0xa8
 
goroutine 1 [sleep]:
github.com/flynn/flynn/pkg/attempt.(*Attempt).Next(0xc208114600, 0xc2080c94a0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/pkg/attempt/attempt.go:55 +0x158
github.com/flynn/flynn/pkg/attempt.Strategy.Run(0x8bb2c97000, 0xbebc200, 0x0, 0xc2081bf2b0, 0x0, 0x0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/pkg/attempt/attempt.go:25 +0x78
main.(*LibvirtLXCBackend).ConfigureNetworking(0xc208033ba0, 0x0, 0xc2080d4f60, 0x20, 0x1, 0x0, 0x0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/host/libvirt_lxc_backend.go:213 +0x36a
main.(*manifestRunner).runManifest(0xc2081193b0, 0x7f1495e4bd80, 0xc20804c428, 0x0, 0x0, 0x0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/host/manifest.go:275 +0x6cc
main.runDaemon(0xc2080a80e0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/host/host.go:244 +0x16d2
github.com/flynn/flynn/host/cli.Run(0x7fff95d43f38, 0x6, 0xc2080a9440, 0x2, 0x2, 0x0, 0x0)
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/host/cli/cli.go:57 +0x4e4
main.main()
	/tmp/tmp.aGMTW3jFSs/src/github.com/flynn/flynn/host/host.go:125 +0x8c1
 
goroutine 6 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35
 
goroutine 17 [syscall, 3 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1
now=2015-04-01T19:22:17+0000 app=host at=start
```

Source: https://gist.github.com/anonymous/1142dbe84b6f96f57ce2#file-upstart-flynn-host-log-L267

--- a/host/logmux/logmux.go
+++ b/host/logmux/logmux.go
@@ -92,8 +92,10 @@ func (m *LogMux) Close() {
 	case <-m.donec:
 	case <-time.NewTimer(3 * time.Second).C:
 		// logs did not drain to logaggregator in 3 seconds, drain them to the local logger
-		close(m.sc.closec)
-		<-m.donec
+		if m.sc != nil {
+			close(m.sc.closec)
+			<-m.donec
+		}
 	}
 }
 

